project/First-Project/src/main/java/first/project/service/AuthenticationServiceImpl.java

```diff
@@ -89,4 +89,38 @@ public boolean checkPassword(String password)
         }
         return res;
     }
+
+    @Override
+    public int checkLoginEmail(String email)
+    {
+        User u = userRepo.findByEmail(email);
+        if(u!=null){
+            return u.getId();
+        }
+        return -1;
+    }
+
+    @Override
+    public boolean checkLoginPassword(String password, int id)
+    {
+        User u=userRepo.findById(id);
+        if(passwordHashing.passwordEncoder().matches(password,u.getPassword())==true)
+            return true;
+        return false;
+    }
+
+    @Override
+    public ArrayList<String> validateAndLogin(User user)
+    {
+        ArrayList<String> res = new ArrayList<String>();
+        int id=checkLoginEmail(user.getEmail());
+        if(id==-1){
+            res.add("The email is not registered");
+        }else{
```

Comment:
Extract this to single else if block:
`else if(checkLoginPassword(user.getPassword(),id)==false){`